cumulusci/tasks/marketing_cloud/deploy.py

```diff
@@ -37,10 +41,14 @@ def _run_task(self):
             self.logger.error(f"Package zip file not valid: {pkg_zip_file.name}")
             return
 
+        custom_inputs = self.options.get("custom_inputs")
+        if custom_inputs:
+            custom_inputs = json.loads(custom_inputs)
```

Comment:
@davisagli - How do you pass a dict directly as a command line option value? It seems like it would always come in as a string that would need to be deserialized into a dict in python.

I may be misunderstanding, but are you saying that `self.options['custom_inputs']` would result in a dict?

Reply:
Or is this some sort of magic that is handled by `process_list_of_pairs_dict_arg` that you mentioned above?

Reply:
@Br4nd0R the task options might come from several different places

1. CLI. You are right that from here they will always come as a string. That's why we have `process_list_of_pairs_dict_arg` to take care of turning a CLI option like `--custom-inputs companyName:Salesforce,logo:salesforce.png` into a dict `{"companyName": "Salesforce", "logo": "salesforce.png"}`

2. YAML. A custom task could specify the option like this:
  ```
  options:
      custom_inputs:
          companyName: Salesforce
  ```
  In this case the options are already a dict once they are parsed from yaml, and the task doesn't need to do any further processing. `process_list_of_pairs_dict_arg` handles this by simply returning its input if it is already a dict.
  
3. Task options passed to Tubular. The Tubular client constructs the task options as a dict and serializes the entire Job including steps and task options to JSON. The Tubular runner deserializes this and constructs a cci flow with the specified steps and task options. This is similar to number 2 in that the option will already be a dict by the time it is passed to the task.